src/groups/mqb/mqbstat/mqbstat_queuestats.cpp

```diff
@@ -678,38 +656,56 @@ void QueueStatsDomain::setQueueContentRaw(bsls::Types::Int64 messages,
 void QueueStatsDomain::updateDomainAppIds(
     const bsl::vector<bsl::string>& appIds)
 {
-    if (!d_subContexts_mp) {
+    if (appIds.empty()) {
+        d_subContextsLookup.clear();
+        d_subContextsHolder.clear();
         return;  // RETURN
     }
 
-    bdlma::LocalSequentialAllocator<2048> localAllocator;
-
-    // Add subcontexts for appIds that are not already present
-    for (bsl::vector<bsl::string>::const_iterator cit = appIds.begin();
-         cit != appIds.end();
-         ++cit) {
-        if (bsl::find_if(d_subContexts_mp->begin(),
-                         d_subContexts_mp->end(),
-                         ContextNameMatcher(*cit)) ==
-            d_subContexts_mp->end()) {
-            StatSubContextMp subContext = d_statContext_mp->addSubcontext(
-                mwcst::StatContextConfiguration(*cit, &localAllocator));
-            d_subContexts_mp->emplace_back(
-                bslmf::MovableRefUtil::move(subContext));
-        }
-    }
-
-    // Remove subcontexts if appIds are not present in updated AppIds
-    bsl::list<StatSubContextMp>::iterator it = d_subContexts_mp->begin();
-    while (it != d_subContexts_mp->end()) {
-        if (bsl::find(appIds.begin(), appIds.end(), it->get()->name()) ==
-            appIds.end()) {
-            it = d_subContexts_mp->erase(it);
+    bsl::unordered_set<bsl::string> remainingAppIds(
+        appIds.begin(),
+        appIds.end(),
+        appIds.size() * 3,  // initial bucket num
```

Comment:
Is hardcoding to `appIds.size() * 3` better than the default implementation-defined bucket num?

Reply:
Not a critical path, removed it.